services/provider/server/cephblockpool.go

```diff
@@ -0,0 +1,102 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	rookCephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type cephBlockPoolManager struct {
+	client    client.Client
+	namespace string
+}
+
+func newCephBlockPoolManager(cl client.Client, namespace string) (*cephBlockPoolManager, error) {
+	return &cephBlockPoolManager{
+		client:    cl,
+		namespace: namespace,
+	}, nil
+}
+
+func (c *cephBlockPoolManager) EnableBlockPoolMirroring(ctx context.Context, blockPoolName string) error {
+
+	cephBlockPool := &rookCephv1.CephBlockPool{}
+	cephBlockPool.Name = blockPoolName
+	cephBlockPool.Namespace = c.namespace
+
+	_, err := ctrl.CreateOrUpdate(ctx, c.client, cephBlockPool, func() error {
+		cephBlockPool.Spec.Mirroring.Enabled = true
+		cephBlockPool.Spec.Mirroring.Mode = "image"
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to enable mirroring on CephBlockPool resource with name %q. %v", blockPoolName, err)
+	}
+
+	return nil
+
+}
+
+func (c *cephBlockPoolManager) SetBootstrapSecretRef(ctx context.Context, blockPoolName, secretName string, secretData map[string][]byte) error {
+
+	cephBlockPool, err := c.GetBlockPoolByName(ctx, blockPoolName)
+	if err != nil {
+		return err
+	}
+
+	// create the secret
+	bootstrapSecret := &corev1.Secret{}
+	bootstrapSecret.Name = secretName
+	bootstrapSecret.Namespace = c.namespace
+
+	_, err = ctrl.CreateOrUpdate(ctx, c.client, bootstrapSecret, func() error {
+		bootstrapSecret.Data = secretData
+		err := ctrl.SetControllerReference(cephBlockPool, bootstrapSecret, c.client.Scheme())
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create/update the bootstrap secret %q. %v", secretName, err)
+	}
+
+	_, err = ctrl.CreateOrUpdate(ctx, c.client, cephBlockPool, func() error {
+		if cephBlockPool.Spec.Mirroring.Peers == nil {
+			cephBlockPool.Spec.Mirroring.Peers = &rookCephv1.MirroringPeerSpec{SecretNames: []string{secretName}}
+		} else {
+			index := slices.IndexFunc(cephBlockPool.Spec.Mirroring.Peers.SecretNames, func(s string) bool {
+				return s == secretName
+			})
+			if index < 0 {
+				cephBlockPool.Spec.Mirroring.Peers.SecretNames = append(cephBlockPool.Spec.Mirroring.Peers.SecretNames, secretName)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to set bootstrap secret ref on CephBlockPool resource with name %q. %v", blockPoolName, err)
+	}
+
+	return nil
+}
+
+func (c *cephBlockPoolManager) GetBlockPoolByName(ctx context.Context, blockPoolName string) (*rookCephv1.CephBlockPool, error) {
+	blockPoolObj := &rookCephv1.CephBlockPool{}
+	err := c.client.Get(ctx, types.NamespacedName{Name: blockPoolName, Namespace: c.namespace}, blockPoolObj)
```

Comment:
maybe we can use utils to construct the type?
``` suggestion
	blockPool := &rookCephv1.CephBlockPool{}
        blockPool.Name = blockPoolName
        blockPool.Namespace = c.namespace
	err := c.client.Get(ctx, client.ObjectKeyFromObject(blockPool), blockPool)
```
similar at remaining places, it just saves us from constructing a struct.